src/sentry/web/frontend/oauth_revoke.py

```diff
@@ -0,0 +1,155 @@
+
+        if not client_id:
+            return self.error(
+                request=request,
+                name="invalid_client",
+                error_description="client_id parameter not found",
+            )
+
+        if not client_secret:
+            return self.error(
+                request=request,
+                name="invalid_client",
+                error_description="client_secret parameter not found",
+            )
+
+        if not token:
+            return self.error(
+                request=request,
+                name="invalid_request",
+                error_description="token parameter not found",
+            )
+
+        if token_type_hint is not None and token_type_hint not in ["access_token", "refresh_token"]:
+            return self.error(
+                request=request,
+                name="unsupported_token_type",
+                error_description="an unsupported token_type_hint was provided, must be either 'access_token' or 'refresh_token'",
+            )
+
+        try:
+            application = ApiApplication.objects.get(
+                client_id=client_id, client_secret=client_secret, status=ApiApplicationStatus.active
+            )
+        except ApiApplication.DoesNotExist:
+            return self.error(
+                request=request,
+                name="invalid_client",
+                error_description="failed to authenticate client",
+                status=401,
+            )
+
+        token_to_delete: ApiToken = self._get_token_to_delete(
+            token=token,
+            token_type_hint=token_type_hint,
+            application=application,  # an application can only revoke tokens it owns
+        )
+
+        # only delete the token if one was found
+        if token_to_delete:
+            token_to_delete.delete()
+            logger.info(
+                "oauth.revoke-success",
+                extra={
+                    "client_id": client_id,
+                    "application_id": application.id,
+                    # don't log the actual token, just a hash of it
+                    "sha256_provided_token": sha256(token.encode("utf-8")).hexdigest(),
+                    "sha256_access_token": sha256(
+                        token_to_delete.token.encode("utf-8")
+                    ).hexdigest(),
+                    "sha256_refresh_token": sha256(
+                        token_to_delete.refresh_token.encode("utf-8")
+                    ).hexdigest(),
+                    "resource_owner_id": token_to_delete.user.id,
+                },
+            )
+
+        # even in the case of invalid tokens we are supposed to respond with an HTTP 200 per the RFC
+        # See: https://www.rfc-editor.org/rfc/rfc7009#section-2.2
+        return HttpResponse(status=200)
+
+    def _get_token_to_delete(
+        self, token: str, token_type_hint: Optional[str], application: ApiApplication
+    ) -> Optional[ApiToken]:
```

Comment:
nit: we're moving away from `Optional[..]` towards something like `ApiToken | None`. Will be fixed automatically at some point, but just fyi

Reply:
Done. :)